packages/components/src/components/switch/switch.stories.ts

```diff
@@ -0,0 +1,174 @@
+import { waitUntil } from '@open-wc/testing-helpers';
+const { argTypes, parameters } = storybookDefaults('sd-switch');
+const { generateTemplate } = storybookTemplate('sd-switch');
+const { overrideArgs } = storybookHelpers('sd-switch');
+
+export default {
+  title: 'Components/sd-switch',
+  component: 'sd-switch',
+  args: overrideArgs([{ type: 'slot', name: 'default', value: 'Default Slot' }]),
+  argTypes,
+  parameters: {
+    ...parameters,
+    design: {
+      type: 'figma',
+      url: 'https://www.figma.com/file/sFzAnWIy3ck28dF4vv1RnE/Switch?node-id=642%3A1032&mode=dev'
+    }
+  }
+};
+
+/**
+ * Default: This shows sd-switch in its default state.
+ */
+
+export const Default = {
+  render: (args: any) => {
+    return generateTemplate({ args });
+  }
+};
+
+/**
+ * Use the disabled attribute to disable an input checkbox. Clicks will be suppressed until the disabled state is removed
+ */
+
+export const Disabled = {
+  parameters: { controls: { exclude: ['disabled', 'default'] } },
+  render: (args: any) => {
+    return generateTemplate({
+      axis: {
+        x: {
+          type: 'attribute',
+          name: 'disabled',
+          values: [false, true]
+        }
+      },
+      constants: { type: 'attribute', name: 'disabled', value: true },
+      args
+    });
+  }
+};
+
+/**
+ * Use the `required` attribute to mark the element as required. This can be used for form validation purposes.
+ */
+
+export const Required = {
+  parameters: { controls: { exclude: ['required'] } },
+  render: (args: any) => {
+    return generateTemplate({
+      axis: {
+        y: { type: 'attribute', name: 'required' }
+      },
+      args
+    });
+  }
+};
+
+export const Checked = {
+  parameters: { controls: { exclude: ['checked'] } },
+  render: (args: any) => {
+    return generateTemplate({
+      axis: {
+        x: {
+          type: 'attribute',
+          name: 'disabled',
+          values: [false, true]
+        }
+      },
+      constants: { type: 'attribute', name: 'checked', value: true },
+      args
+    });
+  }
+};
+
+/**
+ * Test invalid state inside a form.
+ */
+
+export const Invalid = {
+  parameters: { controls: { exclude: ['required'] } },
+  render: (args: any) => {
+    return html`
+      <form>
+        ${generateTemplate({
+          args,
+          constants: [{ type: 'attribute', name: 'required', value: true }]
+        })}
+        <sd-button style="margin-top: 16px" type="submit">Submit</sd-button>
+      </form>
+    `;
+  },
+  play: async ({ canvasElement }: { canvasElement: HTMLUnknownElement }) => {
+    const el = canvasElement.querySelector('sd-button');
+    await waitUntil(() => el?.shadowRoot?.querySelector('button'));
+    await userEvent.type(el!.shadowRoot!.querySelector('button')!, '{return}', { pointerEventsCheck: 0 });
+  }
+};
+
+/**
+ * Use the `base`, `control--unchecked`, `control--checked`, `checked` and `label` part selectors to customize the checkbox.
```

Comment:
I don't know if I would go with the naming checkbox 🤔 I would Propose to take the name from the component -> switch

Reply:
But switch or switched is not a state like checked. Then you have to say switched-on, switched-off. I can do that.

Reply:
I am talking about the last word of your description:
![image](https://github.com/solid-design-system/solid/assets/98025014/c37a6568-08b5-4a33-8362-7b1ac12493dc)

I would change that text to: "**part selectors to customize the switch.**"

Reply:
The rest with control--unchecked, control--checked is good! :)